dfsclient/src/main/java/org/hpccsystems/dfs/client/CompiledFieldFilter.java

```diff
@@ -213,9 +219,25 @@ public static Object getMaxValueForType(FieldType type) throws Exception
                 {
                     return BigDecimal.valueOf(Double.MAX_VALUE);
                 }
+            case CHAR:
             case STRING:
+            case VAR_STRING:
                 {
-                    return maxStringValue;
+                    if (numCodePoints > maxStringValue.length())
+                    {
+                        // Create a string that is at least numCodePoints long
+                        String maxVal = new String(maxStringValue);
+                        for (int i = numCodePoints; i > 0;)
+                        {
+                            maxVal.concat(maxStringValue);
+                            i -= maxStringValue.length();
+                        }
+                        return maxVal;
+                    }
+                    else
+                    {
+                        return maxStringValue;
+                    }
```

Comment:
The fieldtype enum defines other numerics, boolean and other complex types, have we confirmed they're invalid types for index?

Reply:
Correct, there are limited types that can be created in the key of the index